src/components/App/App.js

```diff
@@ -4,20 +4,32 @@ import Navbar from '../Navbar/Navbar';
 import './App.css';
 import { useState } from 'react';
 import "@fontsource/monoton";
-
+import ErrorGrid from '../AllMovies/AllMovies'
 const App = () => {
 
   // make a movie card, pass in props - return the card structure in JSX
   // iterate through the moviesData array, returning the data as cards, assign props in JSX
   const [onHomeView, setOnHomeView] = useState(true);
   const [onDetailsView, setOnDetailsView] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
   const [movies, setMovies] = useState(movieData.movies);
 
+  const filterMovies = (keyword, movieList) => {
+    return [...movieList].filter(movie => movie.title.toLowerCase().includes(keyword.toLowerCase()))
+  }
+
+  const changeSearch = value => {
+    console.log(searchValue, "before")
+    setSearchValue(value);
+    console.log(searchValue, "after")
+    setMovies(filterMovies(value, movieData.movies));
+  }
+  
   return (
     <div className="App">
       <Navbar />
       <main>
-        {onHomeView && <AllMovies movies={movies}/>}
+        {onHomeView && <AllMovies changeSearch={changeSearch} movies={movies} searchValue={searchValue}/>}
```

Comment:
Question: What happens when the movies are not defined? I noticed that you have handled the error in the AllMovies component, but is it also possible to handle it here?